### PR TITLE
Removing Ruby 2.7 testing

### PR DESCRIPTION
**Pull request by johnmccrae**
## Description
Breaking changes in Chef-17 require dropping the rest of support for Ruby 2.7. This is the first of several PR's to update Chef-17 for a new version of Ohai that overcomes an issue with the fqdn suffix on Windows. That update required supporting new versions of gems that do not support Ruby 2.7. That resulted in our testing failing here. We have removed support for Ruby 2.7 previously but were still testing against it. Here we remove the rest of the support. This should not break anything in and of itself. 

This breaks support for Fedora 33. That version supports Ruby 2.7. We'll need to test against Fedora 34 for Ruby 3.0 support

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] If `Gemfile.lock` has changed, I have used `--conservative` to do it and included the full output in the Description above.
- [ ] All new and existing tests passed.
- [ ] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: knife/knife.gemspec [hunk 1]
```diff
@@ -13,7 +13,7 @@ Gem::Specification.new do |s|
   s.email = "adam@chef.io" # These seem a bit out of date, and this address probably doesn't go anywhere anymore?
   s.homepage = "https://www.chef.io"
 
-  s.required_ruby_version = ">= 2.7.0", "< 3.1"
+  s.required_ruby_version = ">3.0.0", "< 3.1"
```
Comment (tpowell-progress) [inline knife/knife.gemspec:16]:
```suggestion
  s.required_ruby_version = ">= 3.0.0", "< 3.1"
```

Comment (tpowell-progress) [reply knife/knife.gemspec:16]: (I know technically `> 3.0.0` should be fine, but I have spun up something recently on `3.0.0` specifically so it seems an unnecessary boundary condition risk. Also... consistency with all the other things and adding a space after the `>=`

**Review by tpowell-progress — CHANGES_REQUESTED** [review]
Small version correction requested.

File: knife/knife.gemspec (hunk 1 above)
Comment (johnmccrae) [reply knife/knife.gemspec:16]: agreed. done

**Review by tpowell-progress — APPROVED** [review]

**Pull request merged**